battle.cpp

```diff
@@ -0,0 +1,143 @@
+#include <bits/stdc++.h>
+#include "main-character.cpp"
+#include "monster-generator.cpp"
+
+using namespace std;
+
+struct in_battle {
+    int consecutive_defending;
+    // The hero in battle also has another attribute to 
+    // check whatever was his last action. But, this
+    // attribute can be derive from the consecutive 
+    // rounds defending: if > 0, last action was a defense
+};
+
+struct hero_in_battle : in_battle {
+    main_character hero;
+};
+
+struct monster_in_battle : in_battle {
+    monster enemy;
+};
+
+int coin_toss() {
+    // Method to decide who attacks first
+    // Input value
+
+    string chosen;
+    cout << "Choose 'heads' or 'tails': ";
+    cin >> chosen;
+    cout << endl;
+
+    // Calculate result
+    // if 1, heads wins
+
+    int result = rand() % 2;
+    if (result == 1 && !chosen.compare("heads")) {
+        return 1; // Hero won
+    } else if (result == 0 && !chosen.compare("tails")) {
+        return 1; // Hero won
+    } 
+    return 0; // Hero lost
+}
+
+int get_battle_action() {
+    int action;
+    cout << "1 for Attack, 2 for Defend, 3 for Super Attack, 4 to use a potion" << endl;
+    cout << "Action: ";
+    cin >> action;
+    cout << endl;
+}
+
+int hero_attacks(hero_in_battle hero, monster enemy) {
+    int action;
+
+    cout << "Choose your action:" << endl;
+    while (1) {
+        action = get_battle_action();
+        if (action == 1) {
```

Comment:
couldn't you use a switch case statement? I think it would be better

Reply:
I actually like if and elses more than switch lets discuss it in the slack